bin/test_runner.rb

```diff
@@ -17,40 +17,53 @@ class Test
   def initialize(test_in_string_format, jclient, rsclient, opts)
     @jclient = jclient
     @rsclient = rsclient
+    @opts = opts
+    @thresholds = @opts[:thresholds]
+
     elems = test_in_string_format.split(' ')
     @percepts = {}
     if elems.size == 1
       @name = test_in_string_format.chomp
-      update_percepts
-      if opts[:force_run] == false && @percepts[:job_status] == "success"
-        puts "force_run flag not off, and test passed, going straight to Done".fg 'yellow'
+      @cloud_name = cloud_name(@name)
+      if @thresholds.has_key?(@cloud_name) && (@thresholds[@cloud_name] == 0)
+        puts "Cloud #{@cloud_name} has threshold 0, moving to Done"
         @stage = Done
       else
-        puts "New test, set to Staging".fg 'yellow'
-        @stage = Staging
+        init_percepts
```

Comment:
Mentioned in the method definition, we should just set the hash here and remove the function.